bindings/pyroot_experimental/PyROOT/test/rdataframe_makerdataframe.py

```diff
@@ -0,0 +1,65 @@
+import unittest
+import ROOT
+import numpy as np
+import sys
+
+
+class MakeRDataFrame(unittest.TestCase):
+    """
+    Tests for the MakeRDataFrame feature enabling to read numpy arrays
+    with RDataFrame.
+    """
+
+    dtypes = [
+        "int32", "int64", "uint32", "uint64", "float32", "float64"
+    ]
+
+    def test_dtypes(self):
+        """
+        Test reading different datatypes
+        """
+        for dtype in self.dtypes:
+            data = {"x": np.array([1, 2, 3], dtype=dtype)}
+            df = ROOT.ROOT.RDF.MakeRDataFrame(data)
+            self.assertEqual(df.Mean("x").GetValue(), 2)
+
+    def test_multiple_columns(self):
+        """
+        Test reading multiple columns
+        """
+        data = {}
+        for dtype in self.dtypes:
+            data[dtype] = np.array([1, 2, 3], dtype=dtype)
+        df = ROOT.ROOT.RDF.MakeRDataFrame(data)
+        colnames = df.GetColumnNames()
+        # Test column names
+        for dtype in colnames:
+            self.assertIn(dtype, self.dtypes)
+        # Test mean
+        for dtype in self.dtypes:
+            self.assertEqual(df.Mean(dtype).GetValue(), 2)
+
+    def test_refcount(self):
+        """
+        Check refcounts of associated PyObjects
+        """
+        data = {"x": np.array([1, 2, 3], dtype="float32")}
+        self.assertEqual(sys.getrefcount(data), 2)
+        self.assertEqual(sys.getrefcount(data["x"]), 2)
+        df = ROOT.ROOT.RDF.MakeRDataFrame(data)
+        self.assertTrue(hasattr(df, "__data__"))
+        self.assertEqual(sys.getrefcount(data["x"]), 3)
```

Comment:
could you add a test in which you do `del data` before the event loop?

Reply:
See above:

> Just tested, we loose the data dictionary, however, the actual numpy arrays survive. Still, if the RDF main node goes out of scope, the data dies. So the ownership of the data is not passed to the last node such as done in C++. But I don't think we can fix this.

Reply:
So from my point of view, we should not increase the refcount for the dictionary itself but only for the internal data. But TBD.